kubeadm/flannel/flannel-overlay.yml

```diff
@@ -17,20 +17,37 @@ data:
     mkdir -force /host/k/flannel
     mkdir -force /host/k/flannel/var/run/secrets/kubernetes.io/serviceaccount
 
-    $cniJson = get-content /etc/kube-flannel-windows/cni-conf.json | ConvertFrom-Json
+    $containerRuntime = "docker"
+    if (Test-Path /host/etc/cni/net.d/0-containerd-nat.json) {
+      $containerRuntime = "containerd"
+    }
+
     $serviceSubnet = yq r /etc/kubeadm-config/ClusterConfiguration networking.serviceSubnet
     $podSubnet = yq r /etc/kubeadm-config/ClusterConfiguration networking.podSubnet
     $networkJson = wins cli net get | convertfrom-json
 
-    $cniJson.delegate.policies[0].Value.ExceptionList = $serviceSubnet, $podSubnet
-    $cniJson.delegate.policies[1].Value.DestinationPrefix = $serviceSubnet
-    Set-Content -Path /host/etc/cni/net.d/10-flannel.conf ($cniJson | ConvertTo-Json -depth 100)
+    if ($containerRuntime -eq "docker") {
+      $cniJson = get-content /etc/kube-flannel-windows/cni-conf.json | ConvertFrom-Json
+
+      $cniJson.delegate.policies[0].Value.ExceptionList = $serviceSubnet, $podSubnet
+      $cniJson.delegate.policies[1].Value.DestinationPrefix = $serviceSubnet
```

Comment:
Why isn't the additional `$cniJson.delegate.policies[2].Value.DestinationPrefix = $networkJson.AddressCIDR` needed for overlay?

Reply:
I'm not sure but I haven't seen that used in any other examples when configuring an overlay network.
Ex: https://github.com/microsoft/SDN/blob/3e54e29aa41bf751bc8a5f528a466b285b4b0291/Kubernetes/windows/helper.v2.psm1#L789-L810

Reply:
In Overlay, local pods talk to local host via HostRoute (which is a policy applied on the network itself by Flannel, developed specifically for Overlay). Only L2Bridge needs the SDN Route to management IP. Not sure on the specifics of how it works though

Reply:
Thanks for the clarity.  Should we add a comment here for others in the future?

Reply:
I think we should document that but i'm not sure this inline'd powershell script in a deployment file is the appropriate place.